presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1049,27 +1050,68 @@ public void testFromInPredicate()
         result = fromPredicate(originalExpression);
         assertEquals(result.getRemainingExpression(), TRUE_LITERAL);
         assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(C_COLOR, Domain.create(ValueSet.of(COLOR, COLOR_VALUE_1, COLOR_VALUE_2).complement(), false))));
+    }
+
+    @Test
+    public void testInPredicateWithNull()
+    {
+        Expression originalExpression = in(C_BIGINT, Arrays.asList(1L, 2L, null));
+        ExtractionResult result = fromPredicate(originalExpression);
+        assertEquals(result.getRemainingExpression(), TRUE_LITERAL);
+        assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(C_BIGINT, Domain.create(ValueSet.ofRanges(Range.equal(BIGINT, 1L), Range.equal(BIGINT, 2L)), false))));
+
+        originalExpression = not(in(C_BIGINT, Arrays.asList(1L, 2L, null)));
+        result = fromPredicate(originalExpression);
+        assertEquals(result.getRemainingExpression(), TRUE_LITERAL);
+        assertEquals(result.getTupleDomain(), TupleDomain.none());
+
+        originalExpression = isNull(in(C_BIGINT, Arrays.asList(1L, 2L, null)));
```

Comment:
wrapping in `isNull` doesn't make much sense as we ask for `C_BIGINT` domain. I wonder why it doesn't throw any error...